### ISSUE TITLE
[v1.5.2] FiltersNestedStack create fails when DestinationArn uses Fn::Join 

### ISSUE THREAD

**Issue by rkdavidson** [dougmoscrop/serverless-plugin-split-stacks#41]
I suspect this split-stacks plugin is to blame, but it could be `dougmoscrop`'s other plugin at fault.
If so, I can move/repost this issue there.

## TL;DR

There is a new issue with either this plugin or [serverless-plugin-log-subscription](https://www.npmjs.com/package/serverless-plugin-log-subscription)'s ability to process `Fn::Join` based values for SubscriptionFilter resource's `destinationArn` and `roleArn` properties. 

## Issue Details
I've got an existing app using [serverless-plugin-log-subscription](https://www.npmjs.com/package/serverless-plugin-log-subscription) and this split-stacks plugin, and when trying to re-deploy recently, the deploy fails when trying to create a new `FiltersNestedStack`:

```CloudFormation - CREATE_FAILED - AWS::CloudFormation::Stack - FiltersNestedStack```

Previously, the `split-stacks` logic resulted in two nested stacks: `APINestedStack` and `PermissionsNestedStack`.

The last successful deploy's stack summary looked like:
```
Serverless: [serverless-plugin-split-stacks]: Summary: 46 resources migrated in to 2 nested stacks
Serverless: [serverless-plugin-split-stacks]:    Resources per stack:
Serverless: [serverless-plugin-split-stacks]:    - (root): 89
Serverless: [serverless-plugin-split-stacks]:    - APINestedStack: 29
Serverless: [serverless-plugin-split-stacks]:    - PermissionsNestedStack: 17
```

I added a new lambda APIGateway endpoint, so not sure if that changed the logic to decide somehow that a third, nested stack for the filters was necessary?

Regardless, I started seeing this error in the serverless deploy results (some sensitive data removed):
```
Serverless: Operation failed!

  Serverless Error ---------------------------------------

  An error occurred: FiltersNestedStack - Embedded stack arn:aws:cloudformation:us-west-2:{AWS_ACCOUNT_ID}:stack/{ROOT_STACK_NAME}-FiltersNestedStack-11M30I1UE4Y79/ac9aa190-650f-11e8-a65a-50a68d01a6c5 was not successfully created: The following resource(s) failed to create: [LIST_OF_ALL_SUBSCRIPTIONFILTER_RESOURCES]. .
```

Checking `CloudFormation` on AWS Console, I saw a bit more detail... next to each SubscriptionFilter, this error was listed:
```
10:43:23 UTC-0700 | CREATE_FAILED | AWS::Logs::SubscriptionFilter | GetRewardInventorySubscriptionFilter | Value of property DestinationArn must be of type String
```

The config for my `logSubscription` plugin is:
```
logSubscription:
  enabled: true
  destinationArn:
    Fn::Join:
      - ':'
      - - 'arn:aws:kinesis:us-west-2'
        - Ref: 'AWS::AccountId'
        - 'stream/splunk-cw-stream'
  roleArn:
    Fn::Join:
      - ':'
      - - 'arn:aws:iam:'
        - Ref: 'AWS::AccountId'
        - 'role/splunk-cw-role'
```

The processed `.json` file for the `FiltersNestedStack` template in `.serverless` directory had the proper JSON equivalent, but seemed a bit strange. It processed the value for the first SubscriptionFilter resource, then all subsequent resources just reference that value.
( Again, edited a bit for sensitive project-based data )
Notice the "$ref" value for the second resource's destinationArn:
```
// .serverless/cloudformation-template-Filters-nested-stack.json
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "FiltersNestedStack nested stack",
  "Parameters": {},
  "Resources": {
    "AuthorizerFuncSubscriptionFilter": {
      "Type": "AWS::Logs::SubscriptionFilter",
      "Properties": {
        "DestinationArn": {
          "Fn::Join": [
            ":",
            [
              "arn:aws:kinesis:us-west-2",
              {
                "Ref": "AWS::AccountId"
              },
              "stream/splunk-cw-stream"
            ]
          ]
        },
        "FilterPattern": "",
        "LogGroupName": "/aws/lambda/{stack}-authorizerFunc",
        "RoleArn": {
          "Fn::Join": [
            ":",
            [
              "arn:aws:iam:",
              {
                "Ref": "AWS::AccountId"
              },
              "role/splunk-cw-role"
            ]
          ]
        }
      },
      "DependsOn": []
    },
    "CheckinDashgetSubscriptionFilter": {
      "Type": "AWS::Logs::SubscriptionFilter",
      "Properties": {
        "DestinationArn": {
          "$ref": "$[\"Resources\"][\"AuthorizerFuncSubscriptionFilter\"][\"Properties\"][\"DestinationArn\"]"
        },
        "FilterPattern": "",
        "LogGroupName": "/aws/lambda/{stack}-checkin-get",
        "RoleArn": {
          "$ref": "$[\"Resources\"][\"AuthorizerFuncSubscriptionFilter\"][\"Properties\"][\"RoleArn\"]"
        }
      },
      "DependsOn": []
    },
    ...
```

To troubleshoot, I hardcoded the final ARN value I wanted into the `logSubscription` config, suspecting an issue with the AWS functions to assemble the final string.
After re-deploy with a string literal value for `destinationArn`, I got the same error for all SubscriptionFilter resources, but this time complaining that `Value of property RoleArn must be of type String`.
I repeated the hard-coding step for `roleArn`, and the deploy was **successful**.

The final, new successful deploy stack summary (with hard coded string values for `destinationArn` and `roleArn`) is:
```
Serverless: [serverless-plugin-split-stacks]: Summary: 68 resources migrated in to 3 nested stacks
Serverless: [serverless-plugin-split-stacks]:    Resources per stack:
Serverless: [serverless-plugin-split-stacks]:    - (root): 76
Serverless: [serverless-plugin-split-stacks]:    - APINestedStack: 32
Serverless: [serverless-plugin-split-stacks]:    - FiltersNestedStack: 18
Serverless: [serverless-plugin-split-stacks]:    - PermissionsNestedStack: 
```

## Not sure what a good workaround/solution is...
I tried using a `Fn::Sub` solution instead, but got this error:

```
Invalid variable reference syntax for variable AWS::AccountId. You can only reference env vars, options, & files. You can check our docs for more info.
```

I guess I _could_ set a temporary option variable that references the AccountId, but that seems hacky.
I'd much prefer the `Fn::Join` call works as it used to.

Any ideas?

**Comment by dougmoscrop** [comment]
Could you confirm what version of serverless, and these plugins, as well as any other plugins you have installed are?

The $ref syntax is unfamiliar to me and looks like something is going wrong inside of serverless variable resolution at first glance.

**Comment by dougmoscrop** [comment]
Actually it looks like it comes from `json-cycle` - let me see if I can reproduce it.

**Comment by rkdavidson** [comment]
Thanks for quick response, Doug.
Apologies for not including that info originally.

Here's the plugins I'm using:
```
plugins:
  - serverless-dynamodb-local
  - serverless-s3-local
  - serverless-offline
  - serverless-offline-scheduler
  - serverless-plugin-log-subscription
  - serverless-plugin-split-stacks
```

Using node `v6.10.0`.

Attached is the result of `npm list`
[package-list.txt](https://github.com/dougmoscrop/serverless-plugin-split-stacks/files/2060177/package-list.txt)

**Comment by dougmoscrop** [comment]
I've reproduced it minimally with just the log-sub plugin, opened another issue; closing this one!
